pkg/controller/etcdproxy/etcdproxy_controller.go

```diff
@@ -383,19 +421,21 @@ func (c *Controller) handleObject(obj interface{}) {
 	}
 }
 
-// newDeployment creates a new Deployment for a EtcdStorage resource. It also sets
+// newReplicaSet creates a new Deployment for a EtcdStorage resource. It also sets
 // the appropriate OwnerReferences on the resource so handleObject can discover
 // the EtcdStorage resource that 'owns' it.
-func newDeployment(etcdstorage *samplev1alpha1.EtcdStorage) *appsv1.Deployment {
+func (c *Controller) newReplicaSet(etcdstorage *samplev1alpha1.EtcdStorage) *appsv1.ReplicaSet {
 	labels := map[string]string{
-		"app":        "nginx",
-		"controller": etcdstorage.Name,
+		"controller": "epc",
 	}
-	repls := int32(3)
-	return &appsv1.Deployment{
+	replicas := int32(1)
+	etcdaddr := fmt.Sprintf("--endpoints=%s", c.etcdConnectionInfo.EtcdURL)
+	etcdns := fmt.Sprintf("--namespace=/%s", etcdstorage.ObjectMeta.Name)
+
+	return &appsv1.ReplicaSet{
 		ObjectMeta: metav1.ObjectMeta{
-			Name:      "test",
-			Namespace: etcdstorage.Namespace,
+			Name:      fmt.Sprintf("etcd-rs-%s", etcdstorage.ObjectMeta.Name),
```

Comment:
private function for this name computation since you need to match in both spots.

Reply:
Added and documented `getReplicaSetName` function.